compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4990,10 +4998,18 @@ object Types {
           foldOver(cs + sym, tp)
         case tp: TypeRef if tp.info.isTypeAlias =>
           apply(cs, tp.superType)
-        case tp: TypeBounds =>
-          foldOver(cs, tp)
-        case other =>
+        case tp: TypeRef if tp.prefix.isValueType =>
```

Comment:
What does the `prefix.isValueType` condition accomplish?

Reply:
It was a half-baked attempt at `tp.typeSymbol.isClass`.